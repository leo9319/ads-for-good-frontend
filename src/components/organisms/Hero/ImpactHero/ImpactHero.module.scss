@import '../../../../styles/colours';
@import '../../../../styles/breakpoints';
@import '../../../../styles/spacing';
@import '../../../../styles/typography';

@mixin fontProperties(
  $size,
  $line,
  $weight: 600,
  $align: center,
  $color: $color-neutral-950
) {
  font-size: var($size);
  line-height: var($line);
  font-weight: $weight;
  text-align: $align;
  color: $color;
}

@mixin flexProperties($align, $direction, $padding, $gap) {
  display: flex;
  align-items: $align;
  align-self: stretch;
  flex-direction: $direction;
  justify-content: start;
  padding: var($padding);
  gap: var($gap);
}

.impactHeroBox {
  @include flexProperties(
    $align: center,
    $direction: column,
    $padding: --spacing-2xs,
    $gap: --spacing-lg
  );
  width: 100%;
  max-width: 872px;
  margin: auto;
  padding: var(--spacing-3xl) 0;

  @media (max-width: $md-max) {
    padding: var(--spacing-lg) var(--spacing-2xl);
  }

  @media (max-width: $xs-max) {
    padding: var(--spacing-lg) var(--spacing-md);
  }

  .header {
    gap: var(--spacing-xs);

    .caption {
      @include fontProperties(
        $size: --typography-font-size-10,
        $line: --typography-line-height-10
      );

      @media (max-width: $md-max) {
        @include fontProperties(
          $size: --typography-font-size-8,
          $line: --typography-line-height-8
        );
      }

      @media (max-width: $xs-max) {
        @include fontProperties(
          $size: --typography-font-size-7,
          $line: --typography-line-height-7
        );
      }
    }

    .headeline {
      gap: var(--spacing-2xs);

      .title {
        color: $color-orange-600;
      }

      .subtitle {
        color: $color-neutral-950;
      }
    }

    .description {
      @include fontProperties(
        $size: --typography-font-size-3,
        $line: --typography-line-height-3,
        $weight: 400,
        $color: $color-neutral-600
      );
    }
  }

  .body {
    width: 100%;

    .peopleImage {
      width: 100% !important;
      margin: auto;
      border-radius: 16px;

      @media (min-width: $lg-min) {
        width: 100% !important;
        max-width: 760px !important;
      }
    }

    &_card {
      @include flexProperties(
        $align: center,
        $direction: column,
        $padding: --spacing-md,
        $gap: --spacing-md
      );
      border-radius: 16px;
      border: 1px solid $color-neutral-100;
      background: $color-white;

      &_label {
        @include fontProperties(
          $size: --typography-font-size-5,
          $line: --typography-line-height-5,
          $color: $color-neutral-600
        );
      }

      &_value {
        @include fontProperties(
          $size: --typography-font-size-5,
          $line: --typography-line-height-5,
          $color: $color-neutral-950
        );
      }
    }

    .bottomSection {
      padding: var(--spacing-md) !important;

      .titleSection {
        padding-bottom: var(--spacing-sm) !important;
      }

      .bottomTitle {
        font-weight: var(--typography-font-weight-semibold) !important;
        font-size: var(--typography-font-size-5) !important;
        line-height: var(--typography-line-height-5) !important;
        color: $color-neutral-950;
      }

      .bottomAmount {
        font-weight: var(--typography-font-weight-semibold) !important;
        font-size: var(--typography-font-size-7) !important;
        line-height: var(--typography-line-height-7) !important;
        color: $color-orange-600;
      }

      .bottomDesc {
        font-weight: var(--typography-font-weight-regular) !important;
        font-size: var(--typography-font-size-3) !important;
        line-height: var(--typography-line-height-3) !important;
        color: $color-neutral-600;
      }
    }
  }
}
