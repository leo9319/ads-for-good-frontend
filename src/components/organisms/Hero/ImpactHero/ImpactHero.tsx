import React, { ReactElement } from 'react';
import Flex from '@radix-styles/atoms/Flex';
import Text from '@radix-styles/atoms/Text';
import Box from '@radix-styles/atoms/Box';
import Image from '@components/atoms/Image';
import { classNames } from '@utils/common/classNames';
import FallbackImage from '@assets/images/FallbackImage.svg';

import styles from './ImpactHero.module.scss';

/**
 * @interface
 * The ImpactHero props type of {@link ImpactHero}.
 */
export interface ImpactHeroProps {
  /**
   * Main title displayed in the hero section.
   * @property
   */
  title: string;

  /**
   * Subtitle displayed under the main title.
   * @property
   */
  subTitle: string;

  /**
   * Description text below the title and subtitle.
   * @property
   */
  description: string;

  /**
   * Image URL for the hero section.
   * If unavailable, a fallback image will be used.
   * @property
   */
  ImpactHeroImage: string;

  /**
   * An optional parameter to modify the existing card styles
   * @property
   */
  className?: string;

  /**
   * An optional parameter to display bottom title
   * @property
   */
  bottomTitle?: string;

  /**
   * An optional parameter to display bottom amount
   * @property
   */
  bottomAmount?: string;

  /**
   * An optional parameter to display bottom description
   * @property
   */
  bottomDesc?: string;
}

/**
 * ImpactHero component
 *
 * @group Organism
 * @category Component
 *
 * @param {ImpactHeroProps} props - The ImpactHero props
 * @returns {ReactElement} - ImpactHero Component
 */
export const ImpactHero = ({
  title,
  subTitle,
  description,
  ImpactHeroImage,
  className,
  bottomTitle,
  bottomAmount,
  bottomDesc,
}: ImpactHeroProps): ReactElement => {
  return (
    <Box className={classNames(styles.impactHeroBox, className)}>
      <Flex display="flex" direction="column" className={styles.header}>
        <Flex display="flex" direction="column" className={styles.headeline}>
          <Text className={classNames(styles.caption, styles.title)}>
            {title}
          </Text>
          <Text className={classNames(styles.caption, styles.subtitle)}>
            {subTitle}
          </Text>
        </Flex>
        <Text className={styles.description}> {description} </Text>
      </Flex>
      <Flex display="flex" direction="column" className={styles.body}>
        <Image
          src={ImpactHeroImage}
          alt="Impact Hero Image"
          fallbackSrc={FallbackImage}
          className={styles.peopleImage}
        />
        <Flex
          display="flex"
          direction="column"
          className={styles.bottomSection}
        >
          <Box className={styles.titleSection}>
            <Text
              as="p"
              className={classNames(styles.body_card_label, styles.bottomTitle)}
            >
              {bottomTitle}
            </Text>
            <Text
              as="p"
              className={classNames(
                styles.body_card_value,
                styles.bottomAmount
              )}
            >
              {bottomAmount}{' '}
            </Text>
          </Box>
          <Box>
            <Text
              as="p"
              className={classNames(styles.body_card_value, styles.bottomDesc)}
            >
              {bottomDesc}{' '}
            </Text>
          </Box>
        </Flex>
      </Flex>
    </Box>
  );
};

export default ImpactHero;
