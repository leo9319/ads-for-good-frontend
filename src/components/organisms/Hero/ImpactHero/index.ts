export type { ImpactHeroProps } from './ImpactHero';
export { ImpactHero } from './ImpactHero';
export { default } from './ImpactHero';
