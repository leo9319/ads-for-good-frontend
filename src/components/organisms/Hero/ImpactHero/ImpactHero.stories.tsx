import ImpactHero from './ImpactHero';
import { StoryObj, Meta } from '@storybook/react';

import PeopleHeroImage from '@assets/images/PeopleHero.svg';

const meta = {
  title: 'Library/Organisms/Hero/ImpactHero',
  component: ImpactHero,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    title: { control: 'text' },
    subTitle: { control: 'text' },
    description: { control: 'text' },
    ImpactHeroImage: { control: 'text' },
    className: { control: 'text' },
    bottomDesc: { control: 'text' },
    bottomTitle: { control: 'text' },
    bottomAmount: { control: 'text' },
  },
} satisfies Meta<typeof ImpactHero>;

export default meta;

type Story = StoryObj<typeof meta>;

export const PersonHeroOrganism: Story = {
  args: {
    title: '1',
    subTitle: 'person reached globally',
    description: 'You’re not just donating—you’re transforming lives.',
    ImpactHeroImage: PeopleHeroImage,
    bottomDesc:
      'Did you know: for every $50-$100 donated, a new life is reached!',
  },
};

export const PeopleHeroOrganism: Story = {
  args: {
    title: '54',
    subTitle: 'people reached globally',
    description: 'You’re not just donating—you’re transforming lives.',
    ImpactHeroImage: PeopleHeroImage,
    bottomDesc: 'For every $50-$100 donated, a new life is reached!',
    bottomTitle: 'My donations this year',
    bottomAmount: '$500',
  },
};
