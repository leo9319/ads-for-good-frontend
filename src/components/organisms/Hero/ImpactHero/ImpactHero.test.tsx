import React from 'react';
import { render } from '@testing-library/react';
import ImpactHero from './ImpactHero';

import ImpactHeroImage from '@assets/images/ImpactHero.svg';

describe('Impact Hero Organism', () => {
  it('Impact Hero Organism should render successfully', () => {
    const { baseElement } = render(
      <ImpactHero
        title="1"
        subTitle="person reached globally"
        description="You’re not just donating—you’re transforming lives."
        ImpactHeroImage={ImpactHeroImage}
        bottomDesc="Did you know: for every $50-$100 donated, a new life is reached!"
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
