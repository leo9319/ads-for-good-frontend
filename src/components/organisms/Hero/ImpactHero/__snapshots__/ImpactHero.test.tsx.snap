// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Impact Hero Organism Impact Hero Organism should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface ImpactHeroBox"
    >
      <div
        class="rt-Flex rt-r-display-flex rt-r-fd-column header"
      >
        <div
          class="rt-Flex rt-r-display-flex rt-r-fd-column headeline"
        >
          <span
            class="rt-Text mocked-class-name typography-font-family-default caption title"
          >
            7.6 Million
          </span>
          <span
            class="rt-Text mocked-class-name typography-font-family-default caption subtitle"
          >
            People Reached Globally
          </span>
        </div>
        <span
          class="rt-Text mocked-class-name typography-font-family-default description"
        >
           
          Your support is transforming lives around the world.
           
        </span>
      </div>
      <div
        class="rt-Flex rt-r-display-flex rt-r-fd-column body"
      >
        <img
          alt="Impact Hero Image"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper impactImage"
          inert=""
          src="SvgrURL"
          tabindex="-1"
        />
        <div
          class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface body_card"
        >
          <div
            class="rt-Flex rt-r-display-flex rt-r-fd-column"
          >
            <span
              class="rt-Text mocked-class-name typography-font-family-default body_card_label"
            >
              Our 2025 Goal=
            </span>
            <span
              class="rt-Text mocked-class-name typography-font-family-default body_card_value"
            >
              Reach 7 Million Lives
               
            </span>
          </div>
          <div
            class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface progressBarBox"
          >
            <div
              class="rt-Flex rt-r-display-flex rt-r-fd-row rt-r-jc-space-between"
            >
              <span
                class="rt-Text mocked-class-name typography-font-family-default"
              >
                Overall Progress
              </span>
              <span
                class="rt-Text mocked-class-name typography-font-family-default"
              >
                3.01 million lives
              </span>
            </div>
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="33"
              aria-valuetext="33%"
              class="rt-ProgressRoot rt-r-size-3 rt-variant-surface progressBarContainer progressBar"
              data-max="100"
              data-state="loading"
              data-value="33"
              role="progressbar"
              style="--progress-value: 33;"
            >
              <div
                class="rt-ProgressIndicator"
                data-max="100"
                data-state="loading"
                data-value="33"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
